### ISSUE TITLE
Pan-sahelian keyboard (NFD) needs to reorder diacritics

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyboards#127]
Steps to reproduce:

1. Create bookmarklet for **Fulfulde, Adamawa Keyboard**
```javascript
javascript: void((function() {
    try {
        var e = document.createElement('script');
        e.type = 'text/javascript';
        e.src = 'https://r.keymanweb.com/code/bml20.php?langid=fub&keyboard=ps32&lang=Fulfulde%2C Adamawa pan-Sahelian (v. 3.2)%26font%3D%7B"family"%3A"LatinWeb"%2C"source"%3A%5B"DejaVuSans.ttf"%2C"DejaVuSans.mobileconfig"%5D%2C"files"%3A%5B"DejaVuSans.ttf"%2C"DejaVuSans.mobileconfig"%5D%2C"path"%3A"https%3A%2F%2Fs.keyman.com%2Ffont%2Fdeploy%2F"%7D';
        document.body.appendChild(e);
    } catch (v) {}
})())
```
2. Use the bookmarklet on a website
3. Using the Pan-sahelian keyboard, type
`w` + `2` + `9` ==> `ẃ̩`

The resulting diacritics (U+0301) + (U+0329) are in the wrong order

**Comment by DavidLRowe** [comment]
If you type `w + 9 + 2` are the diacritics in the correct order? Do keyboards have to guarantee that the output is perfectly formed? If so, I suspect there are a lot of keyboards that wouldn't qualify!

**Comment by andjc** [comment]
@darcywong00 does this problem also occurs in windows version? Or just the js version? And which version of pan-sahelian?

**Comment by darcywong00** [comment]
Sorry, I see that I created this issue. But I'm drawing a blank if it was on behalf of someone, or from a legacy issue report.

**Comment by DavidLRowe** [comment]
I think the keyboard is this one: https://github.com/keymanapp/keyboards/tree/master/legacy/p/ps32

**Comment by andjc** [comment]
I had a look at the source to see what is happening.
1) ther eis no js version, was never developed for KMW or bookmarklets.
2) there are ordering constraints for diacritics but these only come into play with the vowels.
3) A sequence such as 
```
w + 2 + 9 ==> ẃ̩
```
was never explicitly intended. Probably easiest fix to to prevent use of 2 or more diacriitcs with conssonants. Will add it to my post contract task list

**Comment by andjc** [comment]
Although layout is probably in need of updated research and a complete overhaul and rewrite. It is quite old now.

**Comment by LornaSIL** [comment]
This is the case in this keyboard: https://keymanweb.com/#fub,Keyboard_el_pan_sahelian
The ps32 keyboard is deprecated. It probably wouldn't be hard to add a group to make it nfc.
However, we've never required that we reorder diacritics in our keyboards.

**Comment by mcdurdin** [comment]
> However, we've never required that we reorder diacritics in our keyboards.

Yes, I think this should be considered a stretch goal at this point: normalized output is definitely preferable, but there are many keyboards (Keyman and otherwise) which never normalize any output.

Ideally, all [Keyman] keyboards should produce fully normalized output. LDML keyboards do automatically generate normalized output (unless the author turns off the guardrails, not recommended).

We still hope to support normalization in .kmn keyboards in the future (keymanapp/keyman#5809) which would also catch this.

**Comment by LornaSIL** [comment]
I think we decided we wouldn't require normalization.
